clang/lib/CIR/CodeGen/CIRGenExprConst.cpp

```diff
@@ -1015,6 +1015,20 @@ class ConstExprEmitter
         return {};
     }
 
+    auto desiredType = CGM.getTypes().ConvertType(T);
+    bool isDesiredArrayOfUnionDirectly = [&]() {
```

Comment:
Can you add some comments explaining the workaround? Might provide some extra insights when comparing to OG.

Reply:
Done